src/insexpand.c

```diff
@@ -1400,7 +1412,24 @@ ins_compl_build_pum(void)
 	if (fuzzy_filter && compl_leader.string != NULL && compl_leader.length > 0)
 	    compl->cp_score = fuzzy_match_str(compl->cp_str.string, compl_leader.string);
 
+	if (is_forward && compl->cp_cpt_source_idx != -1)
+	{
+	    if (cur_source != compl->cp_cpt_source_idx)
+	    {
+		cur_source = compl->cp_cpt_source_idx;
+		match_count = 1;
+		max_matches_found = FALSE;
+	    }
+	    else if (cpt_sources_array && !max_matches_found)
+	    {
+		max_matches = cpt_sources_array[cur_source].max_matches;
```

Comment:
Init here max_matches instead of at the start of the function? It doesn't seem to be used elsewhere.
```suggestion
		int max_matches = cpt_sources_array[cur_source].max_matches;
```